azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/maven/MavenSettingHelper.java

```diff
@@ -0,0 +1,101 @@
+     * Get auth configuration from serverId.
+     *
+     * @param session           the maven session
+     * @param settingsDecrypter the decrypter
+     * @param serverId          the server id
+     * @return the auth configuration
+     * @throws AzureLoginException when there are errors decrypting maven generated password
+     */
+    public static MavenAuthConfiguration getAuthConfigurationFromServer(MavenSession session, SettingsDecrypter settingsDecrypter, String serverId)
+            throws AzureLoginException {
+        if (StringUtils.isBlank(serverId)) {
+            throw new IllegalArgumentException("Parameter 'serverId' cannot be null or empty.");
+        }
+        final Server server = session.getSettings().getServer(serverId);
+        if (server == null) {
+            throw new InvalidConfigurationException(String.format("serverId '%s' cannot be found in maven settings.xml.", serverId));
+        }
+        final MavenAuthConfiguration configurationFromServer = new MavenAuthConfiguration();
+        final Xpp3Dom configuration = (Xpp3Dom) server.getConfiguration();
+        configurationFromServer.setServerId(serverId);
+        if (configuration == null) {
+            return configurationFromServer;
+        }
+        configurationFromServer.setTenant(getConfiguration(configuration, "tenant"));
+        configurationFromServer.setClient(getConfiguration(configuration, "client"));
+        final String rawKey = getConfiguration(configuration, "key");
+        configurationFromServer.setKey(isValueEncrypted(rawKey) ? decryptMavenProtectedValue(settingsDecrypter, "key", rawKey) : rawKey);
+        configurationFromServer.setCertificate(getConfiguration(configuration, "certificate"));
+        final String rawCertificatePassword = getConfiguration(configuration, "certificatePassword");
+        configurationFromServer.setCertificatePassword(isValueEncrypted(rawCertificatePassword) ?
+                decryptMavenProtectedValue(settingsDecrypter, "certificatePassword", rawCertificatePassword) :
+                rawCertificatePassword);
+        configurationFromServer.setEnvironment(getConfiguration(configuration, "environment"));
+        configurationFromServer.setHttpProxyHost(getConfiguration(configuration, "httpProxyHost"));
+        configurationFromServer.setHttpProxyPort(getConfiguration(configuration, "httpProxyPort"));
+        configurationFromServer.setServerId(serverId);
+
+        // validate configuration
+        return configurationFromServer.validateAndReturn();
+    }
+
+    /**
+     * Get string value from server configuration section in settings.xml.
+     *
+     * @param configuration the <code>Xpp3Dom</code> object representing the configuration in &lt;server&gt;.
+     * @param property           the property name
+     * @return String value if property exists; otherwise, return null.
+     */
+    private static String getConfiguration(final Xpp3Dom configuration, final String property) {
```

Comment:
rename to `getValue`/`getPropertyValue`.
`Configuration` should be `MavenAuthConfiguration` in current scope.